Tests/SentryTests/Networking/TestRequestManager.swift

```diff
@@ -19,7 +21,9 @@ public class TestRequestManager: NSObject, RequestManager {
     var responseDelay = 0.0
     public func add( _ request: URLRequest, completionHandler: SentryRequestOperationFinished? = nil) {
         
+        semaphore.wait()
         requests.append(request)
```

Comment:
Where is this being read? You need to synchronize the readers too

Reply:
Good point. Thanks @bruno-garcia